File: Data Structures & Algorithms/two-integer-sum/submission-1.go
func twoSum(nums []int, target int) []int {
	// Init the hashmap
	kv := make(map[int]int)
	for i, v := range nums {
		kv[v] = i
	}

	// Lookup
	for i, v := range nums {
		x := target - v 
		j, ok := kv[x]
		if ok && i != j {
			return []int{i, j}
		}
	}

	return []int{}
}

File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        for (int l = 0, r = nums.length - 1; l <= r; l++, r--) {
            if (nums[l] == target) return l;
            if (nums[r] == target) return r;
        }
        return -1;
    }
}
